### PR TITLE
WT-2566 Lock/unlock operations should imply memory barriers.

### PR DESCRIPTION
**Pull request by michaelcahill**

### CODE REVIEW

File: src/support/mtx_rw.c [hunk 1]
```diff
@@ -316,31 +330,32 @@ __wt_writelock(WT_SESSION_IMPL *session, WT_RWLOCK *rwlock)
 int
 __wt_writeunlock(WT_SESSION_IMPL *session, WT_RWLOCK *rwlock)
 {
-	wt_rwlock_t *l, copy;
+	wt_rwlock_t *l, new;
```
Comment (michaelcahill) [inline src/support/mtx_rw.c]: This variable rename is just for consistency with other functions in the same file.

**Comment by michaelcahill** [comment]
@agorrod, this pull request has just the barrier changes.  I would like to get the into MongoDB asap and these changes are required for WT-2560.  Can you please review?

**Comment by agorrod** [comment]
Thanks @michaelcahill lgtm - I'm happy for this to be merged if tests pass.

**Comment by michaelcahill** [comment]
Thanks, @agorrod, will do.

**Pull request merged**